graphite_api/app.py

```diff
@@ -553,6 +560,8 @@ def tree_json(nodes, base_path, wildcards=False):
 
 
 def doImageRender(graphClass, graphOptions):
+    if CAIRO_DISABLED:
+        return
```

Comment:
An error response is returned earlier L478-481. With an additional check in the `if request_options['graphType'] == 'pie':` branch, you could return an error message there as well and I think this check would not be needed.